src/rewriter.fs

```diff
@@ -416,6 +416,61 @@ module private RewriterImpl =
+    // The reused identifier gets compressed better, and the declaration is sometimes removed.
+    // float d1 = f(); float d2 = g();  ->  float d1 = f(); d1 = g();
+    let reuseExistingVarDecl b =
+        let tryReplaceWithPrecedingAndFollowing f (xs : Stmt list) =
+            let rec go f preceding = function
+                | head :: tail ->
+                    match f (preceding, head, tail) with
+                    | None -> head :: go f (head :: preceding) tail // preceding is reversed, that's good
+                    | Some xs -> xs @ tail
+                | [] -> []
+            go f [] xs
+        b |> tryReplaceWithPrecedingAndFollowing (function
+        | (preceding2, Decl (ty2, declElts), following2) ->
+            let findAssignmentReplacementFor declElt2 =
+                let compatibleDeclElts = preceding2 |> List.collect (function
+                    // The previous declaration must have the same type.
+                    | Decl (ty1, declElts1) when ty2 = ty1 ->
+                        let firstIsNotUsedAfterSecondDeclared (declElt1 : DeclElt) followingDecl2 =
+                            // The first variable must not be used after the second is declared.
+                            Analyzer.varUsesInStmt (Block followingDecl2) |> List.forall (fun i -> i.Name <> declElt1.name.Name)
+                        declElts1 |> List.filter (fun declElt1 ->
+                            // The previous declaration must have the same size and semantics
+                            declElt1.size = declElt2.size &&
+                            declElt1.semantics = declElt2.semantics &&
+                            firstIsNotUsedAfterSecondDeclared declElt1 following2
+                        )
+                    | _ -> [])
+                match compatibleDeclElts |> List.tryHead with
+                | Some (declElt1) ->
+                    debug $"{declElt2.name.Loc}: eliminating local variable '{declElt2.name}' by reusing existing local variable '{declElt1.name}'"
+                    for v in Analyzer.varUsesInStmt (Block following2) do // Rename all uses of var2 to use var1 instead.
+                        if v.Name = declElt2.name.Name then
+                            v.Rename(declElt1.name.Name)
+                    match declElt2.init with
+                    | Some init -> Some [Expr (FunCall (Op "=", [Var declElt1.name; init]))]
+                    | None -> Some []
+                | _ -> None
+            match declElts, (declElts |> List.rev) with
+            | [declElt2], _  -> // float d1=f(); ...; float d2=g();  ->  ...; d1=g();
+                match findAssignmentReplacementFor declElt2 with
+                | Some stmts -> Some stmts // Replace the Decl with the assignment. Largest win
+                | None -> None
+            //| (declElt2 :: others), _ -> // float d1=f(); ...; float d2=g(),d3=h();  ->  ...; d1=g(); float d3=h();
```

Comment:
remove comment block?

Reply:
I'll enable it soon, after I figure out the last bugs